From 171bd0aea0bb18a27b791388f50843fe9fe25eaa Mon Sep 17 00:00:00 2001
From: Arne Coucheron <arco68@gmail.com>
Date: Sun, 9 Oct 2016 06:18:47 +0200
Subject: [PATCH 10/20] stagefright: omx: Don't signal dataspace change on
 legacy QCOM

This isn't supported in legacy media HAL, and causes things like
screen recording and wifi display to fail when setting up the encoder.

This commit squashes:
  stagefright: omx: Rename legacy defaults to match lineage soong variable
  Change-Id: I783f75b2a6aea5b4f5619038888db394028e2010

Change-Id: Icb3f7b7dfefcfd72939037241568f28c01fc11ed
---
 media/libstagefright/bqhelper/Android.bp              | 1 +
 media/libstagefright/bqhelper/GraphicBufferSource.cpp | 2 ++
 2 files changed, 3 insertions(+)

diff --git a/media/libstagefright/bqhelper/Android.bp b/media/libstagefright/bqhelper/Android.bp
index 0e2b47227d..f55b564f29 100644
--- a/media/libstagefright/bqhelper/Android.bp
+++ b/media/libstagefright/bqhelper/Android.bp
@@ -9,6 +9,7 @@ package {
 
 cc_defaults {
     name: "libstagefright_bufferqueue-defaults",
+    defaults: ["stagefright_qcom_legacy_defaults"],
     double_loadable: true,
 
     srcs: [
diff --git a/media/libstagefright/bqhelper/GraphicBufferSource.cpp b/media/libstagefright/bqhelper/GraphicBufferSource.cpp
index cff14ac185..d031e91f5a 100644
--- a/media/libstagefright/bqhelper/GraphicBufferSource.cpp
+++ b/media/libstagefright/bqhelper/GraphicBufferSource.cpp
@@ -891,11 +891,13 @@ status_t GraphicBufferSource::submitBuffer_l(const VideoBuffer &item) {
         return UNKNOWN_ERROR;
     }
 
+#ifndef QCOM_BSP_LEGACY
     if ((android_dataspace)item.mDataspace != mLastDataspace) {
         onDataspaceChanged_l(
                 item.mDataspace,
                 (android_pixel_format)item.mBuffer->getGraphicBuffer()->format);
     }
+#endif
 
     std::shared_ptr<AcquiredBuffer> buffer = item.mBuffer;
     // use a GraphicBuffer for now as component is using GraphicBuffers to hold references
-- 
2.34.1

